api/base/serializers.py

```diff
@@ -143,92 +173,141 @@ def get_attribute(self, obj):
         return field_source_method(auth=auth)
 
 
-class JSONAPIHyperlinkedIdentityField(ser.HyperlinkedIdentityField):
+class RelationshipField(ser.HyperlinkedIdentityField):
     """
-    HyperlinkedIdentityField that returns a nested dict with url,
-    optional meta information, and link_type.
+    RelationshipField that permits the return of both self and related links, along with optional
+    meta information.
 
     Example:
+    children = RelationshipField(
+        related_view='nodes:node-children',
+        related_view_kwargs={'node_id': 'pk'},
+        self_view='nodes:node-node-children-relationship',
+        self_view_kwargs={'node_id': 'pk'},
+        related_meta={'count': 'get_node_count'}
+    )
 
-        children = JSONAPIHyperlinkedIdentityField(view_name='nodes:node-children', lookup_field='pk',
-                                    link_type='related', lookup_url_kwarg='node_id', meta={'count': 'get_node_count'})
+    If you surround the lookup_field in angular brackets, it will not attempt to find that attribute on the target, but rather
```

Comment:
I don't have a strong preference. Steve's argument is persuasive, but in my experience the error that it gives when someone uses it wrong is going to be more important than the docs. I proxy my tiebreaker to @cwisecarver.